src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1132,13 +1133,15 @@ private void Filter_SelectionChanged(object sender, FilterChangedEventArgs e)
                 NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                 {
                     await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    var sw = Stopwatch.StartNew();
                     if (_isTransitiveDependenciesExperimentEnabled)
                     {
                         _packageList.ClearPackageLevelGrouping();
                     }
 
                     await SearchPackagesAndRefreshUpdateCountAsync(useCacheForUpdates: true);
-                    EmitRefreshEvent(timeSpan, RefreshOperationSource.FilterSelectionChanged, RefreshOperationStatus.Success, isUIFiltering: false);
+                    sw.Stop();
+                    EmitRefreshEvent(timeSpan, RefreshOperationSource.FilterSelectionChanged, RefreshOperationStatus.Success, isUIFiltering: false, sw.Elapsed.TotalMilliseconds);
```

Comment:
Can we add the duration everywhere where the #s can be safely calculated?

In particular, this only captures things like change a tab. 
Changing the prerelease checkbox also changes things so worth tracking that as well, as long as it's measurable.

Reply:
I found place where I can capture more scope.
What I tested here: Check/unheck prerelease, enable/disable/reorder selection of sources for all 4 Tabs in Solution PMUI and 3 tabs in Project PMUI.